src/main/java/Duke.java

```diff
@@ -1,10 +1,239 @@
+    }
+
+    // Level-2: adds new task to user
+    public String add(String text) {
+        Task nextTask = new Task(text);
+        taskList.add(nextTask);
+        fileStorage.saveEntry(nextTask);
+        return "added: ".concat(text);
+    }
+
+    // Level-4: setting of task urgency to "to-do"
+    public String todo(String description) {
+        Task task = new Todo(description,TASK_TYPE.T);
+        taskList.add(task);
+        fileStorage.saveEntry(task);
+        return "Got it. I've added task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(taskList.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-4: setting of task urgency to "deadline"
+    // Level-4: setting of task urgency to "deadline"
+    public String deadline(String description, String by) {
+        Task task = new Deadline(description,by,TASK_TYPE.D);
+        taskList.add(task);
+        fileStorage.saveEntry(task);
+        return "Got it. I've added task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(taskList.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-4: setting of task urgency to "event"
+    public String event(String description, String at) {
+        Task task = new Event(description,at,TASK_TYPE.E);
+        taskList.add(task);
+        fileStorage.saveEntry(task);
+        return "Got it. I've added task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(taskList.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-6: delete items from list
+    public String delete(int index) {
+        Task task = taskList.get(index);
+        taskList.remove(index);
+        fileStorage.deleteEntry(task);
+        return "Noted. I've removed task:\n  "
+                .concat(task.toString())
+                .concat("\nNow you have ".concat(taskList.size() + "").concat(" tasks in the list."));
+    }
+
+    // Level-3: Mark items as done
+    public String done(int index) {
+        taskList.get(index).setDone();
+        Task task = taskList.get(index);
+        return "Nice! I've marked task as done: \n  ".concat(task.toString());
+    }
+
+    // Level-2: Lists all items in Task List
+    public String list() {
+        String out = "Here are the tasks in your list:\n";
+        System.out.println(taskList);
+        for (int i = 0; i < taskList.size(); i++) {
+            if (i != 0) out = out.concat("\n");
+            out = out.concat((i+1) + ".").concat(taskList.get(i).toString());
+        }
+        return out;
+    }
+
+    // wraps all messages between line separators
+    public String messageWrapper(String text) {
+        return SEP_LINE.concat("\n").concat(text).concat("\n").concat(SEP_LINE);
+    }
+
+    // decodes input from user and passes that argument to a response builder
+    public int decoder(String userInput) {
```

Comment:
Should this be changed to decodeInput to be more detailed? 
```suggestion
    public int decodeInput(String userInput) {
```